src/components/NavBar/Util.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import {
+  Box,
+  Link,
+  Text,
+  Button,
+  Image,
+  Stack,
+  Flex,
+  Heading,
+} from '@chakra-ui/react';
+import {CloseIcon} from '@chakra-ui/icons';
+import {FaBars, FaGlobeAmericas} from 'react-icons/fa';
+import logo512 from './logo512.png';
+import {useTranslation} from 'react-i18next';
+
+/**
+ * Props for Menu Toggle Button
+ */
+interface MenuToggleProps {
+  toggle: () => void;
+  isOpen: boolean;
+}
+
+/**
+ * Menu toggle button for mobile-sized screens
+ * @param toggle - function that toggles whether the menu is open or closed
+ * @param isOpen - boolean, is the menu open
+ * @returns a button that toggles the menu when clicked
+ */
+const MenuToggle: ({toggle, isOpen}: MenuToggleProps) => JSX.Element = ({
+  toggle,
+  isOpen,
+}: MenuToggleProps) => {
+  return (
+    <Button
+      variant="ghost"
+      _hover={{background: 'teal'}}
+      onClick={toggle}
+      width="60px"
+    >
+      {isOpen ? <CloseIcon size="lg" /> : <FaBars size="lg" />}
+    </Button>
+  );
+};
+
+/**
+ * Props for the Language Toggle
+ */
+interface LanguageToggleProps {
+  toggle: () => void;
+}
+
+/**
+ * Button to change the language between English and Spanish
+ * @param toggle - function that toggles the website language
+ * @returns a button that toggles the language when clicked
+ */
+const LanguageToggle: ({toggle}: LanguageToggleProps) => JSX.Element = ({
+  toggle,
+}: LanguageToggleProps) => {
+  const {t} = useTranslation('menu');
+  return (
+    <Button
```

Comment:
Formatting inconsistent with other links on hover, language only bolds while the others bold and underline